urllib3/util/wait.py

```diff
@@ -117,7 +116,7 @@ def _have_working_poll():
     # from libraries like eventlet/greenlet.
     try:
         poll_obj = select.poll()
-        poll_obj.poll(0)
+        _retry_on_intr(poll_obj.poll, 0)
```

Comment:
Nice catch here.

Reply:
I can't take credit - the unit test exposed it.

Reply:
You can take credit for the unit test though. ;)